File: frontend/src/services/profileService.js
import api from "../api/axios";

/*
----------------------------------
Get Logged-in User Profile
(Student / Staff / Admin)
----------------------------------
*/

export const getProfile = async () => {
  const response = await api.get("/profile");

  return response.data?.data || {};
};

/*
----------------------------------
Update Logged-in User Profile
(Student / Staff / Admin)
----------------------------------
*/

export const updateProfile = async (data) => {
  const payload = {
    name: data.name,
    phone: data.phone,
    department: data.department,
    bio: data.bio,
    profile_image: data.profile_image,
  };

  // Student-only fields
  if (data.roll_number !== undefined) {
    payload.roll_number = data.roll_number;
  }

  if (data.hostel !== undefined) {
    payload.hostel = data.hostel;
  }

  // Staff/Admin fields
  if (data.designation !== undefined) {
    payload.designation = data.designation;
  }

  if (data.office !== undefined) {
    payload.office = data.office;
  }

  const response = await api.put("/profile", payload);

  return response.data?.data || {};
};

File: backend/services/profile_service.py
from database.db import db
from models.user_model import User


class ProfileService:

    # ----------------------------
    # Get Logged-in User Profile
    # ----------------------------

    @staticmethod
    def get_profile(user_id):

        user = User.query.get(user_id)

        if not user:
            return None

        return user.to_dict()

    # ----------------------------
    # Update Profile
    # ----------------------------

    @staticmethod
    def update_profile(user_id, data):

        user = User.query.get(user_id)

        if not user:
            return None

        # Editable Fields

        if "name" in data:
            user.name = data["name"].strip()

        if "phone" in data:
            user.phone = data["phone"].strip()

        if "department" in data:
            user.department = data["department"].strip()

        if "roll_number" in data:
            user.roll_number = data["roll_number"].strip()

        if "hostel" in data:
            user.hostel = data["hostel"].strip()

        if "bio" in data:
            user.bio = data["bio"].strip()

        if "profile_image" in data:
            user.profile_image = data["profile_image"]

        db.session.commit()

        return user.to_dict()